### ISSUE TITLE
Router configuration for Contentful webhook path

### ISSUE THREAD

**Issue by steinitz-zz** [matfin/meteor-contentful#6]
Hi,

I could use some help with creating the route for the contentful webhook.  I can see in meteor.com logs that contentful is hitting the route which I've configured but the logging I've added to your package's listener doesn't occur (and the Contentful database isn't updated).

I've tried several iron router configurations, including: <pre><code>Router.map (
    function()
    {
        this.route (
            'contentfulhook',
            {
                where: 'server',
                path: '/contentfulhook'
            }
        );
    }
);

Router.route (
    '/contentfulhook',
    {
        name: 'contentfulhook'
    }
);</code></pre> It's tricky to debug because, as you would know, it needs to be deployed to a server, in this case meteor.com, to be able to receive the webhook call.

Thanks for any help.

Cheers,

Steve

**Comment by matfin** [comment]
Hey Steve. 

I used a tool called ngrok which opens the needed ports for your local dev environment to the Internet. You can then put the temporary address they give you into the contentful webhook configuration.

Matt

**Comment by steinitz-zz** [comment]
Hi Matt,

Thanks for your reply.  And thanks for sharing your technique for testing locally.  Sounds good.

I'd still vary much appreciate any information you can provide about the iron router command you used to  configure the contentful hook path in your meteor app.  I confess I have little knowledge of iron router outside of making simple paths to render templates.  A working contentful router path example would route would be very helpful.

Thanks again,

Steve

**Comment by matfin** [comment]
Hi Steve,

Iron router is not used to provide webhook functionality. I'm using node express instead. If you look inside the _src/contentful.js file and see the listen function then you will see how this happens. 

Make sure your settings.js file is set up correctly. Shout if you need assistance with that.

Matt

**Comment by steinitz-zz** [comment]
Hi Matt,

Thanks so much for your offer to help with the set up of the contentful package part of our settings.js.  But I think that's set up correctly.  I picked a port out of thin air: 3049, and followed your helpful instructions on setting up the encoded basic auth credentials.  For now I'm assuming that part of the puzzle is ok.

We've set up iron router to return a custom 404 page for any page it doesn't recognize.  I'm concerned that that might interfere with the webhook's authentication etc.  That's why I asked about the iron router setup.  I'll still need to get my head around that side of things.

I suppose if I can't just force it all into place semi-blindly, I might need to step back a little and just play around with getting meteor to respond to webhooks - to get a feel for it.  I'm guessing you did the same.

Thanks again for your help and suggestions.  Your package is a real gift, and with lots of nice engineering, and it makes for a great Meteor CMS solution.

Cheers,

Steve

**Comment by steinitz-zz** [comment]
Hi Matt,

I got the webhook working.  Thanks again for your help and for your generous share.

I was able to get it working once I understood two things:
1. your package doesn't tell contentful what port to use - the port is overt.  For no reason, I had gotten it into my head that your package informed Contentful of the desired port in either MeteorContentful.start() or MeteorContentful.listen().
2. your package 'hard-codes' the webhook path to /hooks/contentful - leaving the user free from such concerns.  Not understanding that led to my questions about iron router config.

A big aid to my understanding was your helpful suggestion to use ngrok.  Thanks again.

All the best,

Steve
